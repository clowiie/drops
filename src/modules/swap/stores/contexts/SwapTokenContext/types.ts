export interface ContextType {
  fromToken?: string
  setFromToken: (token: string) => void
  toToken?: string
  setToToken: (token: string) => void
}
