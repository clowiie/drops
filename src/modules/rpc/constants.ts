export const BSC_RPC_URL = 'https://bsc-dataseed.binance.org'

export const NOT_FOUND_ADDRESS = '0x0000000000000000000000000000000000000000'
