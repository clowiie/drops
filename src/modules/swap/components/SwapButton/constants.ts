export const DISABLE_BUTTON_CLASS =
  'bg-gray-200 text-white rounded-md p-3 text-lg font-semibold pointer-events-none'

export const ACTIVE_BUTTON_CLASS =
  'bg-gray-700 hover:bg-gray-500 text-white rounded-md p-3 text-lg font-semibold text-center'
