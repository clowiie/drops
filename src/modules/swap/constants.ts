export const SWAP_ROOT_PATH = '/'

export const MAXIMUM_UINT256 =
  '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'

export const DEFAULT_GAS_LIMIT = '150000'

export enum InputSwapToken {
  In = 'in',
  Out = 'out',
}
