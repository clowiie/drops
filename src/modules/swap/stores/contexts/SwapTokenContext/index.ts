import { createContext } from 'react'

import { ContextType } from './types'

const SwapTokenContext = createContext<ContextType>({
  setFromToken: () => {},
  setToToken: () => {},
})

export default SwapTokenContext
