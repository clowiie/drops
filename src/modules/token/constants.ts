import Token from 'common/constants/tokens/bsc'
import { TokenType } from 'modules/token/types'

import ALPACALogo from 'modules/token/images/alpaca-logo.png'
import BUNNYLogo from 'modules/token/images/bunny-logo.svg'
import BUSDLogo from 'modules/token/images/busd-logo.svg'
import CAKELogo from 'modules/token/images/cake-logo.svg'
import ETHLogo from 'modules/token/images/eth-logo.svg'

const tokenList = {
  ALPACA: {
    address: Token.Alpaca,
    name: 'Alpaca',
    symbol: 'ALPACA',
    logo: ALPACALogo,
  },
  BUNNY: {
    address: Token.Bunny,
    name: 'Pancake BUNNY',
    symbol: 'BUNNY',
    logo: BUNNYLogo,
  },
  BUSD: {
    address: Token.Busd,
    name: 'BUSD Token',
    symbol: 'BUSD',
    logo: BUSDLogo,
  },
  CAKE: {
    address: Token.Cake,
    name: 'PancakeSwap Token',
    symbol: 'CAKE',
    logo: CAKELogo,
  },
  ETH: {
    address: Token.Eth,
    name: 'Ethereum Token',
    symbol: 'ETH',
    logo: ETHLogo,
  },
} as Record<string, TokenType>

export const TokenList = Object.freeze(tokenList)
