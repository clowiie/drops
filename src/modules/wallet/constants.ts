import { Connector } from 'modules/wallet/types'

import MetaMaskLogo from 'modules/wallet/images/meta-mask-logo.svg'
import WalletConnectLogo from 'modules/wallet/images/wallet-connect-logo.svg'
import MetaMaskController from 'modules/wallet/stores/controllers/MetaMask'
import WalletConnectController from 'modules/wallet/stores/controllers/WalletConnect'

export const BRIDGE_URL = 'https://pancakeswap.bridge.walletconnect.org/'
export const POLLING_INTERVAL = 12000

export enum MetaMaskMethod {
  GetPermissons = 'wallet_getPermissions',
  RequestAccounts = 'eth_requestAccounts',
  SwitchChain = 'wallet_switchEthereumChain',
}

export enum MetaMaskRPCApi {
  ChainChanged = 'chainChanged',
  AccountsChanged = 'accountsChanged',
}

export enum ConnectorId {
  MetaMask = 'metaMask',
  WalletConnect = 'walletConnect',
}

export enum ConnectorName {
  MetaMask = 'MetaMask',
  WalletConnect = 'WalletConnect',
}

export const connectorList = [
  {
    name: ConnectorName.MetaMask,
    logo: MetaMaskLogo,
    connectorId: ConnectorId.MetaMask,
  },
  {
    name: ConnectorName.WalletConnect,
    logo: WalletConnectLogo,
    connectorId: ConnectorId.WalletConnect,
  },
] as Connector[]

export const ConnectorList = Object.freeze(connectorList)

const walletMapper = {
  [ConnectorId.MetaMask]: MetaMaskController,
  [ConnectorId.WalletConnect]: WalletConnectController,
}

export const WALLET_MAPPER = Object.freeze(walletMapper)
