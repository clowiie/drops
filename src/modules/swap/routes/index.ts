import { SWAP_ROOT_PATH } from 'modules/swap/constants'
import SwapPage from 'modules/swap/pages/SwapPage/component'
import SwapRootPage from 'modules/swap/pages/SwapRootPage/component'

const swapPage = {
  path: SWAP_ROOT_PATH,
  exact: true,
  component: SwapPage,
}

const swapRootPage = {
  path: SWAP_ROOT_PATH,
  component: SwapRootPage,
  routes: [swapPage],
}

export default [swapRootPage]
