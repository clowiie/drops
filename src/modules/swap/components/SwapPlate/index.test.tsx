import { render } from '@testing-library/react'

import createContextWrapper from 'test/utils/createContextWrapper'
import mockComponent from 'test/utils/mockComponent'

import SwapTokenContext from 'modules/swap/stores/contexts/SwapTokenContext'

import { InputSwapToken } from 'modules/swap/constants'
import * as SwapPlateType from '.'

describe('SwapPlate', () => {
  const SwapButtonSpy = mockComponent('modules/swap/components/SwapButton')
  const SwapInputSpy = mockComponent('modules/swap/components/SwapInput')

  const useComparePriceSpy = jest.fn()
  jest.doMock('modules/swap/hooks/useComparePrice', () => useComparePriceSpy)

  const useHandleInputSwapSpy = jest.fn()
  const onChangeAmountInSpy = jest.fn()
  const onChangeAmountOutSpy = jest.fn()
  jest.doMock(
    'modules/swap/hooks/useHandleInputSwap',
    () => useHandleInputSwapSpy,
  )

  const { default: SwapPlate } = require('.') as typeof SwapPlateType

  beforeEach(() => {
    useComparePriceSpy.mockReturnValue(100)

    useHandleInputSwapSpy.mockReturnValue({
      amountIn: '10',
      amountOut: '100',
      estimate: InputSwapToken.In,
      onChangeAmountIn: onChangeAmountInSpy,
      onChangeAmountOut: onChangeAmountOutSpy,
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('snapshots', () => {
    it('should macth snapshots with estimate amount out', () => {
      const { container, getByText } = render(<SwapPlate />, {
        wrapper: createContextWrapper(SwapTokenContext, {
          fromToken: 'BUSD',
          setFromToken: () => {},
          toToken: 'CAKE',
          setToToken: () => {},
        }),
      })

      expect(container.firstChild).toMatchSnapshot()
      expect(getByText('Exchange')).toBeDefined()
      expect(getByText('Trade tokens in an instant')).toBeDefined()
      expect(getByText('Price')).toBeDefined()
      expect(SwapInputSpy).toBeCalledWith({
        onChange: onChangeAmountInSpy,
        title: 'From (estimated)',
        to: '?modalGetFromToken=true',
        tokenSymbol: 'BUSD',
        value: '10',
      })
      expect(SwapInputSpy).toBeCalledWith({
        onChange: onChangeAmountOutSpy,
        title: 'To ',
        to: '?modalGetToToken=true',
        tokenSymbol: 'CAKE',
        value: '100',
      })
      expect(SwapButtonSpy).toBeCalledWith({ amountIn: 10 })
    })

    it('should macth snapshots with estimate amount in', () => {
      useHandleInputSwapSpy.mockReturnValue({
        amountIn: '10',
        amountOut: '100',
        estimate: InputSwapToken.Out,
        onChangeAmountIn: onChangeAmountInSpy,
        onChangeAmountOut: onChangeAmountOutSpy,
      })

      const { container, getByText } = render(<SwapPlate />, {
        wrapper: createContextWrapper(SwapTokenContext, {
          fromToken: 'BUSD',
          setFromToken: () => {},
          toToken: 'CAKE',
          setToToken: () => {},
        }),
      })

      expect(container.firstChild).toMatchSnapshot()
      expect(getByText('Exchange')).toBeDefined()
      expect(getByText('Trade tokens in an instant')).toBeDefined()
      expect(SwapInputSpy).toBeCalledWith({
        onChange: onChangeAmountInSpy,
        title: 'From ',
        to: '?modalGetFromToken=true',
        tokenSymbol: 'BUSD',
        value: '10',
      })
      expect(SwapInputSpy).toBeCalledWith({
        onChange: onChangeAmountOutSpy,
        title: 'To (estimated)',
        to: '?modalGetToToken=true',
        tokenSymbol: 'CAKE',
        value: '100',
      })
      expect(SwapButtonSpy).toBeCalledWith({ amountIn: 10 })
    })

    it('should macth snapshots without compare price', () => {
      useComparePriceSpy.mockReturnValue(undefined)

      const { container, getByText } = render(<SwapPlate />, {
        wrapper: createContextWrapper(SwapTokenContext, {
          fromToken: 'BUSD',
          setFromToken: () => {},
          toToken: 'CAKE',
          setToToken: () => {},
        }),
      })

      expect(container.firstChild).toMatchSnapshot()
      expect(getByText('Exchange')).toBeDefined()
      expect(getByText('Trade tokens in an instant')).toBeDefined()
      expect(SwapInputSpy).toBeCalledWith({
        onChange: onChangeAmountInSpy,
        title: 'From (estimated)',
        to: '?modalGetFromToken=true',
        tokenSymbol: 'BUSD',
        value: '10',
      })
      expect(SwapInputSpy).toBeCalledWith({
        onChange: onChangeAmountOutSpy,
        title: 'To ',
        to: '?modalGetToToken=true',
        tokenSymbol: 'CAKE',
        value: '100',
      })
      expect(SwapButtonSpy).toBeCalledWith({ amountIn: 10 })
    })
  })
})
