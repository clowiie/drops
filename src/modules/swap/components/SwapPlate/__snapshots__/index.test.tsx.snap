// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapPlate snapshots should macth snapshots with estimate amount in 1`] = `
<div
  class="w-full flex justify-center items-center"
>
  <div
    class="bg-white w-full max-w-lg shadow-sm rounded-md mx-2 p-4 sm:p-8 divide-y divide-gray-200"
  >
    <div
      class="flex flex-col pb-4"
    >
      <span
        class="text-lg font-semibold"
      >
        Exchange
      </span>
      <span>
        Trade tokens in an instant
      </span>
    </div>
    <div
      class="pt-8 grid gap-y-8"
    >
      <div />
      <div />
      <div
        class="flex justify-between text-sm"
      >
        <span>
          Price
        </span>
        <span>
          100
           
          BUSD
           per 
          CAKE
        </span>
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`SwapPlate snapshots should macth snapshots with estimate amount out 1`] = `
<div
  class="w-full flex justify-center items-center"
>
  <div
    class="bg-white w-full max-w-lg shadow-sm rounded-md mx-2 p-4 sm:p-8 divide-y divide-gray-200"
  >
    <div
      class="flex flex-col pb-4"
    >
      <span
        class="text-lg font-semibold"
      >
        Exchange
      </span>
      <span>
        Trade tokens in an instant
      </span>
    </div>
    <div
      class="pt-8 grid gap-y-8"
    >
      <div />
      <div />
      <div
        class="flex justify-between text-sm"
      >
        <span>
          Price
        </span>
        <span>
          100
           
          BUSD
           per 
          CAKE
        </span>
      </div>
      <div />
    </div>
  </div>
</div>
`;

exports[`SwapPlate snapshots should macth snapshots without compare price 1`] = `
<div
  class="w-full flex justify-center items-center"
>
  <div
    class="bg-white w-full max-w-lg shadow-sm rounded-md mx-2 p-4 sm:p-8 divide-y divide-gray-200"
  >
    <div
      class="flex flex-col pb-4"
    >
      <span
        class="text-lg font-semibold"
      >
        Exchange
      </span>
      <span>
        Trade tokens in an instant
      </span>
    </div>
    <div
      class="pt-8 grid gap-y-8"
    >
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
